spec/lib/vanagon/project_spec.rb

```diff
@@ -6,6 +6,12 @@
 describe 'Vanagon::Project' do
   let(:component) { double(Vanagon::Component) }
   let(:configdir) { '/a/b/c' }
+  let(:platform) do
+    OpenStruct.new(:settings => {})
+  end
+  let(:upstream_platform) do
+    OpenStruct.new(:settings => {})
+  end
```

Comment:
What's the purpose of having both of these, since they're the same?

Reply:
Ruby makes shallow copies of objects so we need two platform hashes for use in stubbing to load the settings correctly.

Reply:
I'm not sure what shallow copy means / how it's relevant here. Are you saying that if we had a single platform object, the setting would perpetrate through all the examples? Is that bad?

Reply:
The shallow copy means settings changes in any one place would persist for all of them. In the code we make a new platform so we don't overwrite things unintentionally (https://github.com/puppetlabs/vanagon/pull/632/files#diff-5fa3b3d3ed3dd87653614299456ab680R759). In the tests though we stub out the load_platform/load_project calls so we need to have two separate platforms we can use to test that variables are being overwritten properly

Reply:
ookay, yeah, I also wasn't understanding what `platform` was originally referring to here https://github.com/puppetlabs/vanagon/pull/632/files#diff-5fa3b3d3ed3dd87653614299456ab680L759 , or how it's being used..